spec/lib/page_cloner_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+describe PageCloner do
+  let!(:tag)  { create(:tag) }
+  let(:campaign) { create(:campaign) }
+  let(:page)  { create(:page, tags: [tag], campaign: campaign) }
+  let!(:link) { create(:link, page: page) }
+
+  subject(:cloned_page) { PageCloner.clone!(page) }
+
+  it 'clones page' do
+    expect(cloned_page.id).not_to eq(page.id)
+  end
+
+  it 'clones links' do
+    expect(cloned_page.links).not_to eq(page.links)
+    expect(cloned_page.links.count).to eq(1)
```

Comment:
@Tuuleh I compare count to 1 to assert that the associations actually exist. If `page.links` were `nil` `expect(cloned_page.links).not_to eq(page.links)` would still pass, and comparing zero against zero wouldn't catch that. I generally try and compare against a real value.